src/charts/donut/donut-pct-chart-directive.js

```diff
@@ -0,0 +1,354 @@
+     };
+
+     $scope.availData = {
+         'used': '350',
+         'total': '1000'
+     };
+     $scope.pctConfig = {
+       'units': 'GB',
+       'thresholds':{'warning':'60','error':'90'}
+     };
+
+     $scope.pctData = {
+         'used': '350',
+         'total': '1000'
+     };
+     $scope.noneConfig = {
+       'units': 'GB',
+       'thresholds':{'warning':'60','error':'90'}
+     };
+
+     $scope.noneData = {
+         'used': '350',
+         'total': '1000'
+     };
+
+     $scope.custConfig = {
+       'units': 'MHz',
+       'thresholds':{'warning':'60','error':'90'},
+       "legend":{"show":true},
+       'tooltipFn': function (d) {
+         return '<span class="donut-tooltip-pf"style="white-space: nowrap;">' +
+                   d[0].value + ' ' + d[0].name +
+                '</span>';
+       },
+       'centerLabelFn': function (scope) {
+         return '<tspan dy="0" x="0" class="donut-title-big-pf">' +
+                  scope.data.available + '</tspan>' +
+                '<tspan dy="20" x="0" class="donut-title-small-pf">' +
+                  'Free' +
+                '</tspan>';
+       }
+     };
+
+     $scope.custData = {
+         'used': '670',
+         'total': '1000'
+     };
+
+   };
+ </file>
+ </example>
+ */
+angular.module('patternfly.charts').directive('pfDonutPctChart', ['ChartsMixin', '$timeout',
+  function(chartsMixin, $timeout) {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        id: '@',
+        config: '=',
+        data: '=',
+        centerLabel: '@'
+      },
+      replace: true,
+      templateUrl: 'charts/donut/donut-pct-chart.html',
+      controller: ['$scope',
+        function($scope) {
+
+          $scope.donutChartId = 'donutChart';
+          if ($scope.id) {
+            $scope.donutChartId = $scope.id + $scope.donutChartId;
+          }
+
+          $scope.updateAvailable = function(){
+            $scope.data.available = $scope.data.total - $scope.data.used;
+          };
+
+          if ($scope.data.available === undefined) {
+            $scope.updateAvailable();
+          }
+
+          $scope.getStatusColor = function(used, thresholds) {
+            if (!thresholds){
+              return '#0088CE';
+            } else {
+              if (used >= thresholds.error) {
+                return '#CC0000';
+              } else if (used >= thresholds.warning) {
+                return '#EC7A08';
+              } else {
+                return '#0088CE';
+              }
+            }
+          };
+
+          $scope.statusDonutColor = function(scope) {
+            var color = {
+              pattern: []
+            };
+            var percentUsed = scope.data.used / scope.data.total * 100.0;
+            color.pattern[0] = $scope.getStatusColor(percentUsed, scope.config.thresholds);
+            color.pattern[1] = '#D1D1D1';
+            return color;
+          };
+
+          var donutTooltip = function(scope) {
```

Comment:
In this case I'm talking about `var donutTooltip = function(scope) {`.  I personally like having those at the top and the `$scope` functions at the bottom but regardless the `var`s should be declared at the top (since they will be hoisted there anyway).